src/core/sagas/authSaga.ts

```diff
@@ -1,33 +1,106 @@
+import { ITokens } from './../../types/user';
+import { sendLoginDatasSuccessAction } from './../actions/loginAction';
+import { AuthService } from './../../services/AuthService';
 import { IRegistrationState } from './../reducers/registrationReducer';
 import { getRegistrationSelector } from './../selectors/registrationSelector';
+import { getLoginSelector } from "../selectors/loginSelector";
 import { ACTIONS } from './../actions/constants';
-import { sendRegistrationDataError, setUserNameAction } from './../actions/registrationAction';
-import { IUserAuth } from "../../../src/types/user";
+import { sendRegistrationDataErrorAction, } from './../actions';
+import { sendLoginDataErrorAction } from "../actions/loginAction";
+import { IActivationPayload, IUserAuth, IUserLoginAuth } from "../../../src/types/user";
 import { call, put, select, takeEvery } from "@redux-saga/core/effects";
 import { Action } from 'redux-actions';
 
-
 function* sendRegistrationSaga({
     payload: { username, password, email },
 }: Action<IUserAuth>) {
     try {
         const data: IRegistrationState = yield select(getRegistrationSelector); //альтернатива useSelector
         console.log("registrationData:", { username, password, email });
+        yield put(sendRegistrationDataErrorAction(null));
 
-        //yield call() =>
-        //AuthService.singUp({
-        //    username,
-        //    password,
-        //    email
-        // put для очистки полей после регистрации})
-        yield put(setUserNameAction("")); //вставить после
+        yield call(() =>
+            AuthService.registration({
+                username,
+                password,
+                email,
+            })
+        );
+    } catch (e: any) {
+        const error = Object.keys(e.response.data).reduce((acc: string, field: string) => {
```

Comment:
put this in helper 

`const error = Object.keys(e.response.data).reduce((acc: string, field: string) => {
            const value = e.response.data[field];
            return acc + value.join(" ");
        }, '');`